### PR TITLE
Add null Application Property support

### PR DESCRIPTION
**Pull request by bikamani**
Adding changes to fix github [issue](https://github.com/Azure/azure-iot-sdk-csharp/issues/919)

AMQP drops null app properties while MQTT and HTTP do not. All protocols should be consistent in their implementation.
Fixes: #919

### CODE REVIEW

File: e2e/test/EventHubTestListener.cs [hunk 1]
```diff
@@ -133,7 +134,7 @@ private static bool VerifyKeyValue(string checkForKey, string checkForValue, IDi
             {
                 if (checkForKey == key)
                 {
-                    if ((string)properties[checkForKey] == checkForValue)
```
Comment (abhipsaMisra) [inline e2e/test/EventHubTestListener.cs]: You don't want to return true for all `null` values. You should instead pass `null` into `checkForValue` for application properties

File: iothub/device/src/Transport/AmqpIoT/AmqpIoTMessageConverter.cs [hunk 2]
```diff
@@ -339,7 +339,7 @@ private static bool TryGetNetObjectFromAmqpObject(object amqpObject, MappingType
             netObject = null;
             if (amqpObject == null)
             {
-                return false;
```
Comment (abhipsaMisra) [inline iothub/device/src/Transport/AmqpIoT/AmqpIoTMessageConverter.cs]: Is this `amqpObject` specific to application properties for a message only? Is it being passed from any other method? If yes, then we are introducing new behavior where any `amqpObject` can now be `null`.

Comment (bikamani) [reply iothub/device/src/Transport/AmqpIoT/AmqpIoTMessageConverter.cs]: Nope. I see only one reference for messages

**Pull request merged**